src/main/webapp/pipe.js

```diff
@@ -371,6 +374,32 @@ function triggerRebuild(taskId, project, buildId) {
     });
 }
 
+function triggerBuild(taskId) {
+    Q("#rebuild-" + taskId).hide();
+
+    var before;
+    if (crumb.value != null && crumb.value != "") {
+        console.info("Crumb found and will be added to request header");
+        before = function(xhr){xhr.setRequestHeader(crumb.fieldName, crumb.value);}
+    } else {
+        console.info("Crumb not needed");
+        before = function(xhr){}
+    }
+
+    Q.ajax({
+        url: rootURL + "/job/" + taskId + '/build?delay=0sec',
```

Comment:
@dawidmalina Have tested with folder and it dont work. The problem is to calculate the url to do the build request. 
One suggestion is to include the path to the first job in the component and use that as when sending the build request.